griddedio/GriddedIO.F90

```diff
@@ -442,6 +474,87 @@ subroutine CreateVariable(this,itemName,rc)
 
      end subroutine CreateVariable
 
+     function calculate_mae(nsd) result(mae)
+
+        ! This function is based on Table 3 of doi 10.5194/gmd-12-4099-2019
+        ! The algorithm is weird, but it does duplicate the table
+
+        implicit none
+        integer, intent(in) :: nsd
+        real(kind=REAL32) :: mae
+        real(kind=REAL32) :: mae_base
+        integer :: correction
+
+        mae_base = 4.0 * (1.0/16.0)**floor(real(nsd)/2.0) * (1.0/8.0)**ceiling(real(nsd)/2.0)
+
+        if (nsd > 2 .and. mod(nsd, 2) == 0) then
+           correction = 2
+        else if (nsd == 7) then
+           correction = 2
+        else
+           correction = 1
+        end if
```

Comment:
Would love to eliminate the ELSE's here.   Usually one could use SELECT CASE, but that does not quite work here.   In theory one could do:
```
   correction = 2
   if (.not. ((nsd > 2 .and. mod(nsd,2) == 0) .or. nsd == 7) ) then
      correction = 1
   end if
```
But this is a dubious improvement.

Reply:
Hmm. Interesting. I'm not sure what is better. I guess we could go the other way:
```fortran
correction = 1
if ( (nsd > 2 .and. mod(nsd, 2) == 0) .or. nsd == 7 ) then
   correction = 2
end if
```